src/Widgets/MultitaskingView/WindowClone.vala

```diff
@@ -512,59 +513,18 @@ public class Gala.WindowClone : ActorTarget, RootTarget {
         return this;
     }
 
-    /**
-     * When we cross an IconGroup, we animate to an even smaller size and slightly
-     * less opacity and add ourselves as temporary window to the group. When left,
-     * we reverse those steps.
-     */
-    private void drag_destination_crossed (Clutter.Actor destination, bool hovered) {
-        var icon_group = destination as IconGroup;
-        var insert_thumb = destination as WorkspaceInsertThumb;
-
-        // if we have don't dynamic workspace, we don't allow inserting
-        if (icon_group == null && insert_thumb == null
-            || (insert_thumb != null && !Meta.Prefs.get_dynamic_workspaces ())) {
-                return;
-        }
-
-        // for an icon group, we only do animations if there is an actual movement possible
-        if (icon_group != null
-            && icon_group.workspace == window.get_workspace ()
-            && window.is_on_primary_monitor ()) {
-                return;
-        }
-
-        var scale = hovered ? 0.4 : 1.0;
-        var opacity = hovered ? 0 : 255;
-        uint duration = hovered && insert_thumb != null ? insert_thumb.delay : 100;
-        duration = Utils.get_animation_duration (duration);
-
-        window_icon.save_easing_state ();
-
-        window_icon.set_easing_mode (Clutter.AnimationMode.LINEAR);
-        window_icon.set_easing_duration (duration);
-        window_icon.set_scale (scale, scale);
-        window_icon.set_opacity (opacity);
-
-        window_icon.restore_easing_state ();
-
-        if (insert_thumb != null) {
-            insert_thumb.set_window_thumb (window);
-        }
-
-        if (icon_group != null) {
+    private void destination_crossed (Clutter.Actor destination, bool hovered) {
+        if (destination is Meta.WindowActor) {
             if (hovered) {
-                icon_group.add_window (window, false, true);
+                WindowDragProvider.get_instance ().notify_enter (window.get_id ());
             } else {
-                icon_group.remove_window (window, false);
+                WindowDragProvider.get_instance ().notify_leave ();
             }
         }
+    }
```

Comment:
Can we reverse the if statement here to avoid nesting?

```
private void destination_crossed (Clutter.Actor destination, bool hovered) {
    if (!(destination is Meta.WindowActor)) {
        return;
    }

    if (hovered) {
        WindowDragProvider.get_instance ().notify_enter (window.get_id ());
    } else {
        WindowDragProvider.get_instance ().notify_leave ();
    }
}
```